util/failover.go

```diff
@@ -129,169 +116,93 @@ func GetPod(clientset *kubernetes.Clientset, deploymentName, namespace string) (
+	// the specific instance as, as well as which node it is deployed on
+	instanceNodeMap := createInstanceNodeMap(pods)
 
-	// Get PG version
-	var version int
+	// Now get the statistics about the current state of the replicas, which we
+	// can delegate to Patroni vis-a-vis the information that it collects
+	// We can get the statistics about the current state of the managed instance
+	// From executing and running a command in the first pod
+	pod := pods.Items[0]
 
-	rows, err := conn.Query("SELECT current_setting('server_version_num')")
+	// Execute the command that will retrieve the replica information from Patroni
+	commandStdOut, _, err := kubeapi.ExecToPodThroughAPI(
+		request.RESTConfig, request.Clientset, instanceInfoCommand,
+		pod.Spec.Containers[0].Name, pod.Name, request.Namespace, nil)
 
+	// if there is an error, return. We will log the error at a higher level
 	if err != nil {
-		log.Errorf("Could not perform query for version: %s", target)
-		return nil, err
+		return response, err
 	}
 
-	defer rows.Close()
+	// parse the JSON and plast it into instanceInfoList
+	var rawInstances []instanceReplicationInfoJSON
+	json.Unmarshal([]byte(commandStdOut), &rawInstances)
 
-	for rows.Next() {
-		if err := rows.Scan(&version); err != nil {
-			return nil, err
+	// We need to iterate through this list to format the information for the
+	// response
+	for _, rawInstance := range rawInstances {
+		// if this is a primary, skip it
+		if rawInstance.Type == instanceReplicationInfoTypePrimary {
+			continue
 		}
-	}
-	// Get replication info
-	var replicationInfoQuery string
-	var recvLocation uint64
-	var replayLocation uint64
-
-	if version < 100000 {
-		replicationInfoQuery = fmt.Sprintf(
-			replInfoQueryFormat,
-			locationDiffV9, recvV9,
-			locationDiffV9, replayV9,
-		)
-	} else {
-		replicationInfoQuery = fmt.Sprintf(
-			replInfoQueryFormat,
-			locationDiffV10, recvV10,
-			locationDiffV10, replayV10,
-		)
-	}
-
-	rows, err = conn.Query(replicationInfoQuery)
-
-	if err != nil {
-		log.Errorf("Could not perform replication info query: %s", target)
-		return nil, err
-	}
 
-	defer rows.Close()
-
-	for rows.Next() {
-		if err := rows.Scan(&recvLocation, &replayLocation); err != nil {
-			return nil, err
+		// set up the instance that will be returned
+		instance := InstanceReplicationInfo{
+			ReplicationLag: rawInstance.ReplicationLag,
+			Status:         rawInstance.State,
+			Timeline:       rawInstance.Timeline,
 		}
-	}
-
-	return &ReplicationInfo{
-		ReceiveLocation: recvLocation,
-		ReplayLocation:  replayLocation,
-		Node:            "",
-		DeploymentName:  "",
-	}, nil
-}
 
-func getSecrets(clientset *kubernetes.Clientset, cluster *crv1.Pgcluster, namespace string) ([]msgs.ShowUserSecret, error) {
+		// get the instance name that is recognize by the Operator, which is the
+		// first part of the name
+		instance.Name = instanceNamePattern.FindStringSubmatch(rawInstance.PodName)[1]
```

Comment:
FWIW, it seems like this might be worthy of being it's own utility function, something like `GetInstanceName(...)`?

Reply:
This is very specific in deriving the instance name from the Pods that is being output by Patroni. If we need it some more, we could consider moving it into its own utility function.

Reply:
Sounds good.